src/URIs.jl

```diff
@@ -380,6 +380,45 @@ function splitpath(p::AbstractString)
     return elems
 end
 
+"""
+    URIs.normpath(url)
+
+Normalize the path portion of a URI by removing dot segments.
+
+Refer to:
+* https://tools.ietf.org/html/rfc3986#section-5.2.4
+"""
+normpath(url::URI) =
```

Comment:
I feel like this should be `Base.normpath(url::URI)`,
but that does mean renaming the `normpath(p::AbstractString)`
so it won't collide.

Reply:
See the comment below regarding `splitpath`; this module uses Base names but contains them in the URIs namespace. This is just following suit with the other such instances.

Reply:
(It is wrong there, and it is wrong here. But I guess being wrong consistantly is better ...)